pkg/conference/peer_message_processor.go

```diff
@@ -19,27 +19,46 @@ func (c *Conference) processLeftTheCallMessage(participant *Participant, msg pee
 }
 
 func (c *Conference) processNewTrackPublishedMessage(participant *Participant, msg peer.NewTrackPublished) {
-	participant.logger.Infof("Published new track: %s", msg.Track.ID())
+	participant.logger.Infof("Published new track: %s (%s)", msg.TrackID, msg.Layer.String())
 
-	if _, ok := participant.publishedTracks[msg.Track.ID()]; ok {
-		c.logger.Errorf("Track already published: %v", msg.Track.ID())
-		return
+	if track, ok := participant.publishedTracks[msg.TrackID]; !ok {
+		participant.publishedTracks[msg.TrackID] = PublishedTrack{
+			info:   msg.TrackInfo,
+			layers: []peer.SimulcastLayer{msg.Layer},
+		}
+	} else if !funk.Contains(track.layers, msg.Layer) {
+		track.layers = append(track.layers, msg.Layer)
+		participant.publishedTracks[msg.TrackID] = track
 	}
 
-	participant.publishedTracks[msg.Track.ID()] = PublishedTrack{track: msg.Track}
+	// TODO: Oops, that's not very efficient with the simulcast since it means that when 3
+	// layers are published, we will send 3 times the same metadata.
 	c.resendMetadataToAllExcept(participant.id)
 }
 
+func (c *Conference) processRTPPacketReceivedMessage(participant *Participant, msg peer.RTPPacketReceived) {
+	// For now we just forward the lowest layer always and assume that others don't exist.
+	if msg.Layer != peer.SimulcastLayerLow {
```

Comment:
Shouldn't it be this instead? Won't the current version forward all but the low layer?

```suggestion
	if msg.Layer == peer.SimulcastLayerLow {
```

Reply:
Yes, the version of that commit indeed only forwarded the low layer and did not allow switching (so in the first iteration I just wrote some sort of a "drop-in" replacement), in the upcoming commit (poor man's implementation) I handled all layers.